sphinx_immaterial/format_signatures.py

```diff
@@ -0,0 +1,301 @@
+            for i, child in enumerate(node.children):
+                if i != 0:
+                    new_node += sphinx.addnodes.desc_sig_punctuation(",", ",")
+                _append_child_copy_source_info(new_node, child, child)
+            new_node += sphinx.addnodes.desc_sig_punctuation(")", ")")
+            node = new_node
+        if isinstance(node, docutils.nodes.TextElement):
+            assert node.child_text_separator == ""
+            old_children = list(node.children)
+            del node[:]
+            for child in old_children:
+                _extend_children_copy_source_info(node, child, process(child))
+            return [node]
+        raise ValueError("unexpected child")
+
+    node["is_multiline"] = False
+    for child in children:
+        _extend_children_copy_source_info(node, child, process(child))
+
+    if name_text_node_replacement is not None:
+        (  # pylint: disable=unpacking-non-sequence
+            existing_node,
+            new_node,
+        ) = name_text_node_replacement
+        source = existing_node.source
+        line = existing_node.line
+        existing_node.parent.replace(existing_node, new_node)
+        new_node.source = source
+        new_node.line = line
+
+
+class FormatSignaturesTransform(sphinx.transforms.SphinxTransform):
+
+    # Early
+    default_priority = 0
+
+    def apply(self, **kwargs: Any) -> None:
+        formatted_signatures = getattr(self.env, _FORMATTED_SIGNATURES, None)
+        if formatted_signatures is None:
+            return
+        for node in self.document.traverse(sphinx.addnodes.desc_signature):
+            signature_id = node.get(_SIGNATURE_FORMAT_ID)
+            if signature_id is None:
+                continue
+            domain = node.parent["domain"]
+            formatted_signature = formatted_signatures[domain].get(signature_id)
+            if formatted_signature is None:
+                continue
+            _format_signature(node, formatted_signature)
+
+
+def merge_info(
+    app: sphinx.application.Sphinx,
+    env: sphinx.environment.BuildEnvironment,
+    docnames: List[str],
+    other: sphinx.environment.BuildEnvironment,
+) -> None:
+    _get_collected_signatures(env).update(_get_collected_signatures(other))
+
+
+def env_updated(
+    app: sphinx.application.Sphinx, env: sphinx.environment.BuildEnvironment
+) -> None:
+    domain_signatures = _get_collected_signatures(env)
+    domain_formatted_signatures = collections.defaultdict(dict)
+    domain_styles = app.config.clang_format_signatures_domain_styles
+
+    for domain, signatures in domain_signatures.items():
+        source = io.StringIO()
+
+        for sig_id, signature in signatures.items():
+            source.write(f"// {sig_id}\n")
+            source.write(signature.strip().strip(";"))
+            source.write(";\n")
+
+        style = domain_styles[domain]
+        result = subprocess.run(
+            [app.config.clang_format_command, f"-style={style}"],
```

Comment:
Instead of using an f-str, I think you could concatenate the style arg with the `{`/`}` (expected by clang-format's CLI) surrounding the arg's value. This way the user doesn't need to add the curly brackets to a style definition (which follows yaml syntax) in the conf.py option.

There may also be a way to escape curly brackets in an f-str, but I don't know it off-hand.

Reply:
resolving as this discussion is happening on another review comment.